scripts/check_observability_v8_upgrade_continuity.py

```
@@ -352,23 +353,81 @@ def _assert_trace(stamp: str, spans: list[dict[str, Any]]) -> str:
-    if not expected.issubset(observed):
-        missing = sorted(expected - observed)
-        raise ContinuityError(f"Prometheus lost pre/post lifecycle series: {missing}")
+    maximum = dashboards._prometheus_vector(  # noqa: SLF001
+        f"max_over_time({selector}[{window}])",
+        timeout_seconds=60,
+    )
+
+    def by_role(
+        series: list[dict[str, Any]],
+        *,
+        aggregate: Any,
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `aggregate`

(ANN401)